sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1702,12 +2089,335 @@ Exceptions:
+resulting from this command-group submission is associated with.
+
+* Throws synchronously with error code `invalid` if any of `depEvents` is a
+default constructed `dynamic_event` with no underlying SYCL event.
+|===
+
+=== Events
+
+Events can be used with graphs in the following ways:
+
+- Defining dependencies between nodes in the same graph.
+- Defining dependencies between <<inter-graph-dependencies, different graphs>>.
+- Obtaining <<node-execution-events, execution events for invidual nodes>>
+within an executable graph, which can be waited on or used as dependencies for
+eager SYCL operations.
+- Creating external event dependencies for a graph, either
+<<external-event-dependencies, one-time>> or <<dynamic-events, updatable between
+executions>>.
+
+==== Node-Level Execution Events [[node-execution-events]]
+
+Events representing the completion of an individual node inside a given
+executable graph can be obtained using
+`command_graph<graph_state::executable>::get_event(node)`. These events can then
+be waited on or used as dependencies for eager SYCL operations outside of
+graphs. These events may be useful for operations which may be infrequent and
+depend only on some intermediate results of work being done in the graph.
+
+[source, c++]
+....
+sycl::event ExecutionEvent = ExecGraph.get_event(SomeNode);
+
+Queue.submit((sycl::handler& CGH)
+  {
+    CGH.depends_on(ExecutionEvent);
+    CGH.parallel_for(...);
+  });
+
+// The above operation will only execute once SomeNode has finished executing
+// inside ExecGraph
+Queue.ext_oneapi_graph(ExecGraph);
+....
+
+These events represent only the current execution of a given executable graph on
+a device. These events are not unique per execution of a graph. If a graph is
+enqueued multiple times before using one of these events (for example as a
+dependency to an eager SYCL operation or a host wait), it is undefined which
+specific execution of a graph the event will represent. If a dependency on a
+specific graph execution is required this ordering must be enforced by the
+application to ensure there is only a single graph execution in flight when
+using these events.
+
+For example:
+
+[source, c++]
+----
+
+sycl::event NodeExecutionEvent;
+sycl::event GraphCompletionEvent;
+
+for (size_t i = 0; i < GraphIterations; i++){
+  // Obtain the node execution event for the graph
+  NodeExecutionEvent = ExecGraph.get_event(SomeNode);
+
+  // Enqueue the graph for execution
+  sycl::event GraphCompletionEvent = Queue.ext_oneapi_graph(ExecGraph);
+  // Use the event, to wait on the host and perform some intermediate
+  // host-work once that node has completed
+  ExecutionEvent.wait();
+  DoSomethingOnHost();
+  // Wait on the graph finishing to ensure it is complete before the next use of
+  // a node-level execution event
+  GraphCompletionEvent.wait_and_throw();
+}
+----
+
+These events cannot be used to define dependencies between graphs. These should
```

Comment:
does this mean that event from `command_graph<graph_state::executable>::get_event(node)` can not be used to define dependencies between graphs?  If so, we may need to change section 7.2.1 to exclude such event from `regular sycl events`, or change other sections.

In section 7.2.1, `Regular SYCL events: ... and events obtained via command_graph<graph_state::executable>::get_event()`
In section 7.2.2, ` ... or [regular SYCL events] to either handler::depends_on() or property::node::depends_on property, will create external dependencies`
In section 7.2.3, `... or [regular SYCL events] to handler::depends_on() will create external dependencies for graph nodes`

Reply:
Yes that is correct, it is a little clunky to fully specify in this specification because we are adding errors to existing SYCL functions. I've removed them from the definition of regular SYCL events and added emphasis that try to use these for graph nodes is an error (on top of the existing errors in our function definitions).